ui/apps/platform/src/Containers/ComplianceEnhanced/Schedules/components/ReportJobStatusFilter.test.tsx

```diff
@@ -0,0 +1,105 @@
+import React, { useState } from 'react';
+import { render, screen } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import '@testing-library/jest-dom';
+
+import ReportJobStatusFilter, {
+    ensureReportJobStatuses,
+    ReportJobStatus,
+} from './ReportJobStatusFilter';
+
+const getCheckboxOption = (name: string) => {
+    return screen.getByRole('checkbox', {
+        name,
+    });
+};
+
+const ReportJobsStatusFilterWrapper = ({
+    defaultOptions,
+}: {
+    defaultOptions: ReportJobStatus[];
+}) => {
+    const [selectedStatuses, setSelectedStatuses] = useState<ReportJobStatus[]>(defaultOptions);
+    const onChange = (_checked: boolean, value: ReportJobStatus) => {
+        const isStatusIncluded = selectedStatuses.includes(value);
+        const newStatuses = isStatusIncluded
+            ? selectedStatuses.filter((status) => status !== value)
+            : [...selectedStatuses, value];
+        setSelectedStatuses(newStatuses);
+    };
+
+    return <ReportJobStatusFilter selectedStatuses={selectedStatuses} onChange={onChange} />;
+};
+
+describe('ReportJobStatusFilter', () => {
```

Comment:
It's probably debatable if the test is useful now that I think about it. An E2E test would likely be better. The only issue I've found is having to set up the data for the tests. I need to create a schedule, run a report job (which takes time), and then double-check if things filter correctly. It seems like it'll be a bit tricky to test it well. I wonder if it would be better to have the API data already setup so that we can do this. If not, maybe manual tests would suffice. 

Open to comments, suggestions, etc.

Reply:
I think these tests are still useful since they exercise the functionality of the component in isolation. Having e2e tests would be great too, but given the constraints on the long running async job I can see how this might be cumbersome to set up. An alternative if it fits this use case would be to do an e2e test, but intercept and mock the API responses to test the new behavior. I'm not sure if that covers much more than these integration tests though 🤔